game-samples/src/golang/profile-service/models/player.go

```diff
@@ -0,0 +1,185 @@
+package models
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/mail"
+	"time"
+
+	spanner "cloud.google.com/go/spanner"
+	"github.com/google/uuid"
+	"golang.org/x/crypto/bcrypt"
+	iterator "google.golang.org/api/iterator"
+)
+
+type PlayerStats struct {
+	Games_played spanner.NullInt64 `json:"games_played"`
+	Games_won    spanner.NullInt64 `json:"games_won"`
+}
+
+type Player struct {
+	PlayerUUID      string `json:"playerUUID"`
+	Player_name     string `json:"player_name" binding:"required"`
+	Email           string `json:"email" binding:"required"`
+	Password        string `json:"password" binding:"required"`
+	created         time.Time
+	updated         time.Time
+	Stats           spanner.NullJSON `json:"stats"`
+	Account_balance float64          `json:"account_balance"`
```

Comment:
I think this maps to a `NUMERIC` column in Spanner, right? If so, you should use a different type. `float64` and `NUMERIC` are not compatible. The default for Spanner is to use `big.Rat` for NUMERIC.

Reply:
thanks!

Reply:
Is using big.Rat the recommended data type when (eventually) manipulating the value? As opposed to using something like `spanner.NullFloat64`?

Reply:
Yes, if it is a column that is non-nullable. If it is a nullable column, then you should use `spanner.NullNumeric`. But `spanner.NullFloat` and `float64` should not be used with a `NUMERIC` column, only with `FLOAT64` columns. If you mix `NUMERIC` with `float64`, you could run into rounding and precision errors.